wrapper/minecraft/src/main/kotlin/org/quiltmc/qkl/wrapper/minecraft/text/TextStyles.kt

```diff
@@ -0,0 +1,435 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:Suppress("MemberVisibilityCanBePrivate")
+
+package org.quiltmc.qkl.wrapper.minecraft.text
+
+import net.minecraft.entity.Entity
+import net.minecraft.entity.EntityType
+import net.minecraft.item.Item
+import net.minecraft.item.ItemStack
+import net.minecraft.nbt.NbtCompound
+import net.minecraft.text.*
+import net.minecraft.util.Formatting
+import net.minecraft.util.Identifier
+import org.quiltmc.qkl.wrapper.minecraft.text.mixin.StyleAccessor
+import java.util.UUID
+import kotlin.math.roundToInt
+
+/**
+ * Converts RGB values from Integers to a [Color] for use in text.
+ * Each argument should be within 0 and 255. If a number is outside the range, the function will automatically default
+ * it to the maximum value or minimum value, depending on which is nearer to the provided value
+ *
+ * @param red The red channel of the color
+ * @param green The green channel of the color
+ * @param blue The blue channel of the color
+ *
+ * @return A [Color] created from the provided RGB channels
+ *
+ * @author NoComment1105
+ */
+@TextDslMarker
+@Suppress("MagicNumber")
+public fun Color(red: Int, green: Int, blue: Int): Color = Color(
+    (red.coerceIn(0, 255) shl 16) + (green.coerceIn(0, 255) shl 8) + blue.coerceIn(0, 255)
+)
+
+/**
+ * Converts RGB values from floats to a [Color] for use in text.
+ * Each argument should be within 0 and 1. If a number is outside the range, the function will automatically default it
+ * to the maximum value or minimum value, depending on which is nearer to the provided value.
+ *
+ * @param red The red channel of the color
+ * @param green The green channel of the color
+ * @param blue The blue channel of the color
+ *
+ * @return A [Color] created from the provided RGB Channels
+ *
+ * @author NoComment1105
+ */
+@TextDslMarker
+@Suppress("MagicNumber")
+public fun Color(red: Float, green: Float, blue: Float): Color = Color(
```

Comment:
Would it be reasonable to provide an equivalent to this constructor using `Double`? It is Kotlin's default floating point type, so in some cases explicit casts may be required to get floats.

Reply:
Should I do one for longs too while I'm at it?

Reply:
I don't think longs would be passed around too often, and raw numeric literals are of any type which fits them according to the compiler, so unless a numeric literal is really large, the `Int`-based constructors will pick it up

Reply:
I had made a review comment about adding a three-`UByte` constructor method, but I decided against it because `UByte` is never an inferred type (if a literal is marked unsigned, it's `UInt` or `ULong`, whichever can fit it)

Reply:
It only infers `UInt` and `ULong` when there isn't a type provided, but if there is no constructor with `UInt`s or `ULong`s, but only with `UByte`s, then it will make the literals be `UByte`s instead.

Adding a `UByte` constructor, the following seems to compile
`Color(255u, 0u, 255u)`